### DISCUSSION TITLE
DID: Beschreibung und Einheit(en)

### DISCUSSION THREAD

**Discussion by jwende** [open3e/open3e#89] [Ideas]
Moinsen,
Im Zusammenhang mit der Integration in Homeassistant steht gerade die Frage, wo man Beschreibungen und Einheiten der DID's hinterlegt. 
Man kann das in der Konfiguration von HA machen - kein Thema - nur dann wäre der Aufwand nur für die HA Benutzer sichtbar. 
Ich frage mich nun, ob es eher Sinn macht, das hier im open3e zu pflegen und einen vernünftigen Importer für HA zu schreiben ...

**Comment by MyHomeMyData** [comment]
Aus meiner Sicht der richtige Ansatz, das in open3e zu machen.

Ich nutzte die Datenpunktliste von open3e ja auch als Basis für den ioBroker-Adapter. Infos zu Einheiten und Beschreibung könnte ich da evtl. auch nutzen. Momentan erzeuge ich per Tool `Open3E_dids2json.py` aus der Datenpunktliste ein json-File, das ich im Adapter einlese.

Wie würde denn Dein Ansatz konkret aussehen und wie kommen die Infos dann zu HA oder anderen Anwendungen? Kannst Du ein Beispiel für einen Datenpunkt erstellen?

**Reply by jwende** [reply]
Erste Idee:

import json
import yaml
from datetime import date

import Open3Edatapoints

dataIdentifiers = dict(Open3Edatapoints.dataIdentifiers["dids"])

base_topic='open3e/vitocal' 
didsDict = {}
didsYaml = {}

print('Start conversion of datapoints "Open3Edatapoints.py" to yaml format.')

didsListVersion = date.today().strftime("%Y%m%d")

cntDps = 0
cntWrt = 0

didsDict['sensor']=[]

for dp in dataIdentifiers:
    didsElem = {'state_topic': base_topic+'/'+str(dp)+'_'+dataIdentifiers[dp].id,
           'unit_of_measurement': '',
           'name':dataIdentifiers[dp].id,
           'unique_id':dataIdentifiers[dp].id,
           'device_class': 'energy',
           'value_template': ''
           #      '# {{ value_json | float / 1000.0 }}'
           #      'device': { 'name': 'Vitocal 250A',
           #                'identifiers': ['vc250'] 
           #      }
    }
    didsDict['sensor'].append(didsElem)
    cntDps += 1
didsYaml['# Version'] = didsListVersion
didsYaml['mqtt'] = didsDict

with open('Open3Edatapoints.yaml', 'w') as file:
    documents = yaml.dump(didsYaml, file)

print(str(cntDps)+' dids converted to YAML format. See file "Open3Edatapoints.yaml"')
print('Done.')
